File: Libraries/Dispatcher/native/lang-csharp/source/DispatcherHelper.cs
using Interpreter.Structs;
using System.Collections.Generic;

namespace Interpreter.Libraries.Dispatcher
{
    internal static class DispatcherHelper
    {
        public static void FlushNativeQueue(object[] nativeData, List<Value> output)
        {
            if (nativeData != null)
            {
                lock (nativeData[0])
                {
                    List<Value> accumulated = nativeData[1] as List<Value>;
                    if (accumulated != null)
                    {
                        output.AddRange(accumulated);
                        accumulated.Clear();
                    }
                }
            }
        }
    }
}

File: Libraries/Dispatcher/native/lang-java/source/DispatcherHelper.java
package org.crayonlang.libraries.dispatcher;

import java.util.ArrayList;
import org.crayonlang.interpreter.structs.*;

public class DispatcherHelper {

	public static void flushNativeQueue(Object[] nativeData, ArrayList<Value> output) {
		synchronized (nativeData[0]) {
			ArrayList<Value> q = (ArrayList<Value>) nativeData[1];
			for (int i = 0; i < q.size(); ++i) {
				output.add(q.get(i));
			}
			q.clear();
		}
	}

}
